strategy.cpp

```diff
@@ -0,0 +1,252 @@
+ *  5 = Ball in the bottom right part of the image.
+ */
+    pixy.pixyCam(1);
+
+    int caso = -1;
+
+    if (((pixy.ballX > 125) && (pixy.ballX < 150)) && pixy.ballY < 25)
+    {
+        caso = 0;
+    }
+    else if ((pixy.ballX > 135) && (pixy.ballX < 185))
+    {
+        caso = 1;
+    }
+    else if (pixy.ballX > 185)
+    {
+        if (pixy.ballY > 120)
+        {
+            caso = 2;
+        }
+        else if (pixy.ballY < 120)
+        {
+            caso = 4;
+        }
+    }
+    else if (pixy.ballX < 135)
+    {
+        if (pixy.ballY > 120)
+        {
+            caso = 3;
+        }
+        else if (pixy.ballY < 120)
+        {
+            caso = 5;
+        }
+    }
+
+    switch (caso)
+    {
+
+    case -1:
+        while (pixy.ballX == -1)
+        {
+            int m = 0;
+            if (m <= 1)
+            {
+                robot_.setDirection(90);
+                delay(150);
+                robot_.stop();
+                delay(1);
+                pixy.pixyCam(1);
+                m++;
+            }
+            else
+            {
+                robot_.setDirection(270);
+                delay(300);
+                robot_.stop();
+                delay(1);
+                pixy.pixyCam(1);
+                m = 0;
+            }
+        }
+        break;
+
+    case 0:
+        break;
+
+    case 1:
+        robot_.setDirection(0);
+        delay(25);
+        robot_.stop();
+        delay(5);
+        break;
+
+    case 2:
+        robot_.setDirection(315);
+        delay(25);
+        robot_.stop();
+        delay(5);
+        break;
+
+    case 3:
+        robot_.setDirection(45);
+        delay(25);
+        robot_.stop();
+        delay(5);
+        break;
+
+    case 4:
+        robot_.setDirection(225);
+        delay(25);
+        robot_.stop();
+        delay(5);
+        break;
+
+    case 5:
+        robot_.setDirection(135);
+        delay(25);
+        robot_.stop();
+        delay(5);
+        break;
+    }
+}
+
+//Function for the first strategy.
+void Strategy::strategy1()
+{
+ /*
+ * Cases
+ * 1 - The robot doesn't has possession of the ball - 'seguirPelota()' is called
+ * 'seguirPelota()' when the robot gains possession of the ball, the next case is called
+ * 2 - The robot has posession of the ball - The robot will make diagonal front movements depending on its location
+ * on the field. If the goal is to the left of the image, the robot will make a diagonal right movement, and when it reaches
+ * the sideline, it will change to a diagonal left movement.
+ */
+    int cases = 1;
+    bool ballPossession; //Boolean used to check if the robot has possession of the ball.
+    bool onWhiteLine;    //Boolean used to check if the robot is on top of a white line.
+
+    if (ballPossession != false)
+    {
+        cases = 2;
+    }
+
+    switch (cases)
+    {
+
+    case 1:
+        followBall();
+        break;
+
+    case 2:
+        if (pixy.blueGoalX > 140)
```

Comment:
[Magic numbers?](https://en.wikipedia.org/wiki/Magic_number_(programming))
Make some global variables for each pixy value involved so you can change it easily.